### PR TITLE
Add new environment dropin to flannel service

### PR DESCRIPTION
**Pull request by nicklathe**

### CODE REVIEW

File: kubernetes/cloud-configs/controller.yaml [hunk 1]
```diff
@@ -157,16 +157,19 @@ coreos:
       command: start
       enable: true
       drop-ins:
-        - name: 10-etcd.conf
+        - name: 10-environment.conf
+          content: |
+            [Service]
+            Environment="FLANNEL_IMAGE_URL=${flannel_image_repo}"
```
Comment (weitenghuang) [inline kubernetes/cloud-configs/controller.yaml]: where do we use `FLANNEL_IMAGE_URL`?
can this override flanneld.service invocation's [environment value](https://github.com/coreos/coreos-overlay/blob/master/app-admin/flannel/files/flanneld.service#L13)?

File: kubernetes/cloud-configs/controller.yaml [hunk 2]
```diff
@@ -157,16 +157,19 @@ coreos:
...
+            Environment="FLANNEL_IMAGE_TAG=${flannel_image_version}"
```
Comment (weitenghuang) [inline kubernetes/cloud-configs/controller.yaml]: [FLANNEL_VER](https://github.com/coreos/coreos-overlay/blob/master/app-admin/flannel/files/flanneld.service#L12) ?

Comment (dragan) [reply kubernetes/cloud-configs/controller.yaml]: We need to confirm, but I believe you want to be looking at [flannel-wrapper](https://github.com/coreos/coreos-overlay/tree/master/app-admin/flannel-wrapper/files). Which is what we're basing our changes on.

Comment (dragan) [reply kubernetes/cloud-configs/controller.yaml]: Looks like some backwards compatibility changes are in place.

Comment (weitenghuang) [reply kubernetes/cloud-configs/controller.yaml]: You are right!! `FLANNEL_VER environment variable is deprecated, please use FLANNEL_IMAGE_TAG instead`

**Pull request merged**